views.go

```diff
@@ -0,0 +1,167 @@
+package rockset
+
+import (
+	"context"
+	"github.com/rockset/rockset-go-client/openapi"
+	"github.com/rockset/rockset-go-client/option"
+	"github.com/rs/zerolog"
+	"net/http"
+)
+
+// CreateView creates a new view, with an optional description.
+//
+// REST API documentation https://docs.rockset.com/rest-api/#createview
+func (rc *RockClient) CreateView(ctx context.Context, workspace, view, query string,
+	options ...option.ViewOption) (openapi.View, error) {
+	var err error
+	var resp *openapi.CreateViewResponse
+	var hr *http.Response
+
+	q := rc.ViewsApi.CreateView(ctx, workspace)
+	req := openapi.NewCreateViewRequest(view, query)
+
+	opts := option.ViewOptions{}
+	for _, o := range options {
+		o(&opts)
+	}
+
+	if opts.Description != nil {
+		req.Description = opts.Description
+	}
+
+	err = rc.Retry(ctx, func() error {
+		resp, hr, err = q.Body(*req).Execute()
+		return err
+	})
+
+	if err != nil {
+		return openapi.View{}, err
+	}
+
+	log := zerolog.Ctx(ctx)
+	log.Trace().Str("status", hr.Status).Str("state", resp.Data.GetState()).Msg("view created")
+
+	return resp.GetData(), nil
+}
+
+// UpdateView updates an existing view, with an optional description.
+//
+// REST API documentation https://docs.rockset.com/rest-api/#updateview
+func (rc *RockClient) UpdateView(ctx context.Context, workspace, view, query string,
+	options ...option.ViewOption) (openapi.View, error) {
+	var err error
+	var resp *openapi.UpdateViewResponse
+	var hr *http.Response
+
+	q := rc.ViewsApi.UpdateView(ctx, workspace, view)
+	req := openapi.NewUpdateViewRequest(query)
+
+	opts := option.ViewOptions{}
+	for _, o := range options {
+		o(&opts)
+	}
+
+	if opts.Description != nil {
+		req.Description = opts.Description
+	}
+
+	err = rc.Retry(ctx, func() error {
+		resp, hr, err = q.Body(*req).Execute()
+		return err
+	})
+
+	if err != nil {
+		return openapi.View{}, err
+	}
+
+	log := zerolog.Ctx(ctx)
+	log.Trace().Str("status", hr.Status).Msg("view updated")
+
+	return resp.GetData(), nil
+}
+
+// DeleteView deletes a view.
+//
+// REST API documentation https://docs.rockset.com/rest-api/#deleteview
+func (rc *RockClient) DeleteView(ctx context.Context, workspace, view string) error {
```

Comment:
Should document that this is a non-blocking call that just marks the view for deletion and then returns.